cmd/snapd-apparmor/main.go

```diff
@@ -0,0 +1,149 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// This script is provided for integration with systemd on distributions where
+// apparmor profiles generated and managed by snapd are not loaded by the
+// system-wide apparmor systemd integration on early boot-up.
+//
+// Only the start operation is provided as all other activity is managed by
+// snapd as a part of the life-cycle of particular snaps.
+//
+// In addition the script assumes that the system-wide apparmor service has
+// already executed, initializing apparmor file-systems as necessary.
+//
+// NOTE: This script ignores failures in some scenarios as the intent is
+// to simply load application profiles, as many as we can, even if for
+// whatever reason some of those fail.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+	apparmor_interface "github.com/snapcore/snapd/interfaces/apparmor"
+	apparmor_sandbox "github.com/snapcore/snapd/sandbox/apparmor"
+)
+
+// Checks to see if the current container is capable of having internal AppArmor
+// profiles that should be loaded. Callers of this function should have already
+// verified that they're running inside of a container environment with
+// something like `systemd-detect-virt --container`.
```

Comment:
This does not seem to be the case, since this function is executing this check itself.

Reply:
Ah thanks, that's what I get for copy-pasta-ing the comment from the bash version into the Go impl.